tests/foreman/ui/test_template.py

```diff
@@ -310,3 +310,39 @@ def test_positive_clone(self):
                 os_list=os_list,
             )
             self.assertIsNotNone(self.template.search(clone_name))
+
+    @run_only_on('sat')
+    @skip_if_bug_open('bugzilla', 1386334)
+    @tier1
+    def test_positive_advanced_search(self):
+        """Create new provisioning template and associate it with specific
+        organization and location. Also associate it with new hostgroup.
+        Afterwards search for that template by hostgroup
+
+        @id: 5bcecd40-28af-4913-92a4-863c8dc05ecc
+
+        @BZ: 1386334
+
+        @Assert: Template can be found successfully and no error is raised
+        """
+        org = entities.Organization().create()
+        loc = entities.Location().create()
+        hostgroup = entities.HostGroup(
+            organization=[org], location=[loc]).create()
+        template_name = gen_string('alpha')
+        with Session(self.browser) as session:
+            set_context(session, org=org.name, loc=loc.name)
+            make_templates(
+                session,
+                name=template_name,
+                template_path=OS_TEMPLATE_DATA_FILE,
+                custom_really=True,
+                template_type='Provisioning template',
+                hostgroup=hostgroup.name,
+            )
+            self.assertIsNotNone(self.template.search(template_name))
+            try:
+                self.template.search_key = 'hostgroup'
+                self.assertIsNotNone(self.template.search(hostgroup.name))
+            finally:
+                self.template.search_key = 'name'
```

Comment:
Hm... I'm starting thinking adding optional argument `search_key` to `Base.search()` would be a better design. Maybe we should think about it